### ISSUE TITLE
Browser disconnect when calculating media-queries

### ISSUE THREAD

**Issue by JarrodMFlesch** [faceless-ui/window-info#12]
## The Browser Issue

Example breakpoint: `@media (max-width: 850px) { }`

Chrome calculates media-query breakpoints based on the window width.
- If the browser is 850px wide (from edge to edge) the breakpoint will trigger

Safari calculates media-query breakpoints based on the documents body (not including the scrollbar)
- If the browser is 865px wide (from edge to edge) the breakpoint would hit, because it is calculating document body width and not including the scrollbar.

## The window-info Issue

This package follows the chrome approach to calculate when breakpoints have hit, by using the entire window as the measurement. In safari, the disconnect occurs when you are using this package to break at 850px, the JS breakpoint approach will hit **_after_** the browser media-query does, because at 865px in Safari, the 850px media-query styles will apply and the JS ones will not unless the browser shrinks down to 850px.

## Possible Solution

Instead of using the window width to determine if a breakpoint has hit, I think [window.matchMedia](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia) could replace that, so on resize, these functions would run and then they would align with the browsers' stylesheet media-query approach.

Thoughts?

**Comment by jacobsfletch** [comment]
TL;DR
Some browsers (Safari) include the scrollbar width when matching `@media` css rules. This creates a discrepancy between conditional rendering styles in JS and css-based breakpoints.

I am not familiar with `matchMedia` but at first glance could be the path forward here. If we could use CSS as a source of truth (which looks possible) then this is the ticket.

**Comment by JarrodMFlesch** [comment]
Fix has been merged into the `demo` branch and can be seen working properly [here](https://window-info.netlify.app/)

**Comment by jacobsfletch** [comment]
Shipped in `v1.2.9` :boom:

Good work here @JarrodMFlesch
